packages/epo-react-lib/src/layout/SlideoutMenu/SlideoutMenu.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  FunctionComponent,
+  PropsWithChildren,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useOnClickOutside, useKeyDownEvent } from "@/hooks/listeners";
+import MenuContext from "@/contexts/Menu";
+import {
+  StyledOverlay,
+  StyledMenuContainer,
+  StyledMenuHeader,
+  StyledMenuTitle,
+  StyledMenuCallToAction,
+  StyledMenuCloseButton,
+} from "./styles";
+import { MENU_TRANSITION_TIME } from "./constants";
+import IconComposer from "@/svg/IconComposer/IconComposer";
+
+interface SlideoutMenuProps {
+  title: string;
+  callToAction: string;
+  id: string;
+  isOpen: boolean;
+  isSubMenuOpen?: boolean;
+  onOpenCallback?: () => void;
+  onCloseCallback?: () => void;
+}
+
+const SlideoutMenu: FunctionComponent<PropsWithChildren<SlideoutMenuProps>> = ({
+  title,
+  callToAction,
+  id,
+  isOpen = false,
+  isSubMenuOpen = false,
+  onOpenCallback,
+  onCloseCallback,
+  children,
+}) => {
+  const menuRef = useRef<HTMLDivElement>(null);
+  const titleRef = useRef<HTMLHeadingElement>(null);
+  const menuItems = useRef<Set<HTMLButtonElement>>(new Set()).current;
+
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [isHidden, setIsHidden] = useState(!isOpen);
+
+  const isActive = isOpen && !isSubMenuOpen;
+
+  const value = useMemo(
+    () => ({ menuItems, currentIndex }),
+    [menuItems, currentIndex]
+  );
+
+  useEffect(() => {
+    if (isOpen) {
+      setTimeout(firstItem);
+      setIsHidden(false);
+      onOpenCallback && onOpenCallback();
+    } else {
+      setTimeout(() => {
+        setIsHidden(true);
+      }, MENU_TRANSITION_TIME);
+    }
+  }, [isOpen]);
+
+  useEffect(() => {
```

Comment:
this is just watching to see for when the submenu closes so the menuitem that opened the submenu can be refocused